### PR TITLE
perf: cache Intl formatters + add formatDateLong/getMonthName tests

### LINKED ISSUE #103: Cache Intl formatters as module-level constants
## Problem

In `src/lib/format.ts`, `formatCurrency`, `formatDate`, `formatDateLong`, and `getMonthName` all create new `Intl.NumberFormat` or `Intl.DateTimeFormat` instances on every call. `Intl` formatter construction is relatively expensive (parses locale data, resolves options).

## Solution

Cache formatters as module-level constants:

```ts
const eurFormatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'EUR',
})

export const formatCurrency = (cents: number): string =>
  eurFormatter.format(cents / 100)
```

Apply the same pattern to `formatDate`, `formatDateLong`, and `getMonthName`.

## Files

- `src/lib/format.ts`
- `src/__tests__/format.test.ts` (ensure existing tests still pass)

### LINKED ISSUE #114: Add unit tests for formatDateLong and getMonthName
## Problem

`formatDateLong` and `getMonthName` are exported from `src/lib/format.ts` but have zero test coverage in `src/__tests__/format.test.ts`.

## Solution

Add test cases for:

### formatDateLong
- Standard date formatting
- Edge cases: first/last day of month, leap year dates
- Locale-dependent output (use regex matching like existing tests)

### getMonthName
- All 12 months (1-12)
- Full month name output verification

## Files

- `src/__tests__/format.test.ts`

### PR DESCRIPTION
**Pull request by ciampo**
## Summary

- **Cache Intl formatters** (#103): `formatCurrency`, `formatDate`, `formatDateLong`, and `getMonthName` now use module-level cached `Intl.NumberFormat`/`Intl.DateTimeFormat` instances instead of constructing new ones per call.
- **Missing tests** (#114): Added unit tests for `formatDateLong` (boundary dates, leap year) and `getMonthName` (all 12 months, invalid input). All assertions use exact-match `toBe` checks for consistency.
- **Input validation**: `getMonthName` now returns `'—'` for out-of-range or non-integer month values, consistent with `formatDate` and `formatDateLong` which also gracefully degrade for invalid input.

## Test plan

- [x] `pnpm lint` — no lint errors
- [x] `pnpm build` — clean build (includes `tsc --noEmit`)
- [x] `pnpm test:unit` — all 284 tests pass (45 format tests, 7 new)
- [ ] `pnpm test:visual:docker` — no visual regressions (no UI changes)
- [ ] CI: e2e tests pass (runs against deployed preview)

Closes #103, closes #114

### CODE REVIEW

File: src/lib/format.ts [hunk 1]
```diff
@@ -125,12 +127,18 @@ export function toISODateString(d: Date): string {
...
  * @param month - Month number (1-12)
  * @param year - Year
  */
 export const getMonthName = (month: number, year: number): string => {
+  if (!Number.isInteger(month) || month < 1 || month > 12) {
+    return '—'
+  }
   const date = new Date(year, month - 1, 1)
-  return new Intl.DateTimeFormat('en-US', { month: 'long', year: 'numeric' }).format(date)
+  return monthYearFormatter.format(date)
```
[the remark this answers is not in the document]
Comment (ciampo) [reply src/lib/format.ts]: Valid naming observation. The JSDoc now documents the full return format ("January 2026"), which should prevent caller confusion. A rename to `formatMonthYear` would touch multiple call sites and tests — better suited as a follow-up refactor to keep this PR focused on the perf improvement and test coverage.
